src/spatio_temporal_voxel_grid.cpp

```diff
@@ -330,12 +341,38 @@ double SpatioTemporalVoxelGrid::GetFrustumAcceleration( \
 }
 
 /*****************************************************************************/
-void SpatioTemporalVoxelGrid::GetOccupancyPointCloud( \
-                                       pcl::PointCloud<pcl::PointXYZ>::Ptr& pc)
+void SpatioTemporalVoxelGrid::GetOccupancyPointCloud(\
+                                       sensor_msgs::PointCloud2::Ptr& pc2)
 /*****************************************************************************/
 {
-  // return the pointcloud stored
-  pc = _pc;
+  // convert the grid points stored in a PointCloud2
+  sensor_msgs::PointCloud2::Ptr grid_points_cloud (new sensor_msgs::PointCloud2());
+  grid_points_cloud->width  = _grid_points->size();
+  grid_points_cloud->height = 1;
+  grid_points_cloud->is_dense = true;
+
+  sensor_msgs::PointCloud2Modifier modifier(*grid_points_cloud);
+
+  modifier.setPointCloud2Fields(3,
+                                "x", 1, sensor_msgs::PointField::FLOAT32,
+                                "y", 1, sensor_msgs::PointField::FLOAT32,
+                                "z", 1, sensor_msgs::PointField::FLOAT32);
+  modifier.setPointCloud2FieldsByString(1, "xyz");
+
+  sensor_msgs::PointCloud2Iterator<float>iter_x(*grid_points_cloud, "x");
+  sensor_msgs::PointCloud2Iterator<float>iter_y(*grid_points_cloud, "y");
+  sensor_msgs::PointCloud2Iterator<float>iter_z(*grid_points_cloud, "z");
+
+  for(std::list<geometry_msgs::Point32>::iterator it = _grid_points->begin(); it != _grid_points->end(); ++it){
```

Comment:
line length + typedef the list + brackets on a new line

Reply:
Corrected line length + bracket. Not a huge fan of using too much typedef as I can get confused about the the type later. But if you insist, I can do it.

Reply:
I think typedefs make alot of sense if you consistently reuse a templated class over and over but still only use 1 unique template type. I'd agree if we called std::vector<int> vs double, float, Grid, etc but given that the pointcloud is a list of a set type, it makes it clearer to read. Also that way if we want to swap out `list` for `vector` or `map` etc, its a 1 line change ;)